src/classes/STG_InstallScript_TEST.cls

```diff
@@ -244,4 +244,70 @@ public with sharing class STG_InstallScript_TEST {
             System.assertNotEquals(null, addr.Parent_Contact__c);
         }
     }
+
+    @isTest
+    public static void createAdminAccountForContactOnInstall() {
+
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Account_Processor__c = UTIL_Describe.getAdminAccRecTypeID()));
+
+        Contact cont = new Contact(
+                FirstName = 'Test',
+                LastName = 'User',
+                Email = 'usertest@domain.com'
+        );
+
+        ACCT_IndividualAccounts_TDTM.alreadyRunAfterInsert = true; // need to turn off the trigger for this test
+        insert cont;
+        ACCT_IndividualAccounts_TDTM.alreadyRunAfterInsert = false; // back on for install script
+
+
+        Contact contInserted = [Select AccountId from Contact where Id = :cont.Id];
+
+        Test.startTest();
+        Test.testInstall(new STG_InstallScript(), null);
+        Test.stopTest();
+
+        //If the exsiting Contact does not have Account and the default Admin Account record type exists, system should
+        //create Admin account on Install HEDA
+        if (contInserted.AccountId == null && String.isNotBlank(UTIL_Describe.getAdminAccRecTypeID())) {
+            Contact contInserted2 = [Select AccountId from Contact where Id = :cont.Id];
+            System.assertNotEquals(null, contInserted2.AccountId);
+        }
+
```

Comment:
Thanks. The reason that I did not put assertion contInserted.AccountId == null is because in some cases the test cases will fail because the assertion. For example, when user try to Install HEDA in the existing org and the client already has the code/process to insert Account. In this case, when we try to insert Contact, not only our code runs, client's code may also run. And we do not have the control about clients code. If we put assertion to ontInserted.AccountId == null, it is possible it will fail. Any ideas?

Reply:
That makes sense to me. It just feels strange to me that in some scenarios the unit test is not asserting anything...

Reply:
Great conversation here, I would tend to agree with @xinnanlisf here. I'd like to see the test asserting the expected outcomes. All our tests run without seeAllData so it is safe to assume the tests are running in an environment without the Orgs data. @weizsforg is there a way to modify the tests knowing the data is isolated to whatever your test creates?

Reply:
Oh, I think based on Xinnan's comment the assertion is like System.assertEquals(null, contInserted.AccountId) after we insert Contact. If client has the code to insert Account, will this assertion fail?

Reply:
Thanks everyone @xinnanlisf @davidjray. I think I confused myself here. haha. I should put assertion for Contact Account = 0 after insert Contact. The reason is that if somehow it failed, that means something wrong or conflict happen in the org. Even customer have existing code, we should ask them to change it since it will break HEDA logic.

Reply:
After a second thought last night, I still think this might be a bad idea. Ideally, we use unit test to ensure the process is running as expected. If there is a conflict between HEDA and other code, we should be aware of instead of silently handling.

Reply:
That is correct. I also realized that yesterday after David Ray post the comment.